src/coreclr/tools/aot/ILCompiler.ReadyToRun/Interop/IL/Marshaller.ReadyToRun.cs

```diff
@@ -25,8 +25,6 @@ protected static Marshaller CreateMarshaller(MarshallerKind kind)
                     return new BooleanMarshaller();
                 case MarshallerKind.AnsiString:
                     return new AnsiStringMarshaller();
-                case MarshallerKind.UTF8String:
-                    return new UTF8StringMarshaller();
```

Comment:
Should we put it under `#if !READYTORUN` instead of removing?  I assume this code is shared with NativeAOT.

Reply:
This file is not shared

Reply:
this code is specific to R2R. Notice this comment: https://github.com/dotnet/runtime/pull/55624#discussion_r669346533